compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBackend.java

```diff
@@ -108,6 +127,13 @@ public class JBallerinaBackend extends CompilerBackend {
     private DiagnosticResult diagnosticResult;
     private boolean codeGenCompleted;
     private final List<JarConflict> conflictedJars;
+    private final HashMap<String, ByteArrayOutputStream> optimizedJarStreams = new HashMap<>();
+    private final SymbolTable symbolTable;
+    private final UsedBIRNodeAnalyzer usedBIRNodeAnalyzer;
+    protected final HashSet<PackageID> unusedPackageIDs = new HashSet<>();
+    protected final HashSet<PackageId> unusedPackageIds = new HashSet<>();
```

Comment:
Here both variables have a similar name. Shall we change variable names in way that we get an intuition of what it is used for.

Reply:
I think `PackageId` is used in project API level. I will talk with @azinneera and update the decision here.

Reply:
Had a discussion with @azinneera and seems like `PackageId` is used at the project API level to compare old and new instances of the same Package when edit operations are done. And `PackageID` is used during the compilation as an identifier for the packages.

Shall we change the names as follows?

- `unusedPackageIds` --> `unusedProjectLevelPackageIds`
- `unusedPackageIDs` --> `unusedPackageIDs` or `unusedCompilerLevelPackageIds `

What do you think @LakshanWeerasinghe?

Reply:
Seems okay to me. Verify again with @azinneera or @gayaldassanayake.

Reply:
+1 from me for the suggested change. Displays the difference between the two variables.

Reply:
Refactored in 52e86548cc5b2efa895c7658d9e30618479827f1.